products/appengine/api.yaml

```diff
@@ -459,6 +459,800 @@ objects:
+              Endpoints service configuration ID as specified by the Service Management API. For example "2016-09-19r1".
+
+              By default, the rollout strategy for Endpoints is "FIXED". This means that Endpoints starts up with a particular configuration ID.
+              When a new configuration is rolled out, Endpoints must be given the new configuration ID. The configId field is used to give the configuration ID
+              and is required in this case.
+
+              Endpoints also has a rollout strategy called "MANAGED". When using this, Endpoints fetches the latest configuration and does not need
+              the configuration ID. In this case, configId must be omitted.
+          - !ruby/object:Api::Type::Enum
+            name: 'rolloutStrategy'
+            description: |
+              Endpoints rollout strategy. If FIXED, configId must be specified. If MANAGED, configId must be omitted. Default is "FIXED".
+            default_value: :FIXED
+            values:
+              - :FIXED
+              - :MANAGED
+          - !ruby/object:Api::Type::Boolean
+            name: 'disableTraceSampling'
+            description:
+              Enable or disable trace sampling. By default, this is set to false for enabled.
+            default_value: false
+      - !ruby/object:Api::Type::NestedObject
+        name: 'entrypoint'
+        description: |
+          The entrypoint for the application.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'shell'
+            required: true
+            description: |
+              The format should be a shell command that can be fed to bash -c.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'vpcAccessConnector'
+        description: |
+          The entrypoint for the application.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'name'
+            required: true
+            description: |
+              Full Serverless VPC Access Connector name e.g. /projects/my-project/locations/us-central1/connectors/c1.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'automaticScaling'
+        description: |
+          Automatic scaling is based on request rate, response latencies, and other application metrics.
+        exactly_one_of:
+          - automatic_scaling
+          - manual_scaling
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'coolDownPeriod'
+            description: |
+              The time period that the Autoscaler should wait before it starts collecting information from a new instance.
+              This prevents the autoscaler from collecting information when the instance is initializing,
+              during which the collected usage would not be reliable. Default: 120s
+            default_value: "120s"
+          - !ruby/object:Api::Type::NestedObject
+            name: 'cpuUtilization'
+            required: true
```

Comment:
I made this `required` since it was the only one of these fields with a default.  My other option was to put an `AtLeastOneOf` on all of these fields and set a default here, which I can do.  It was more of a maintenance decision to do it this way rather than user-centric decision.